src/hooks/useWalletBalances.ts

```diff
@@ -25,23 +25,30 @@ export type WalletBalanceResult = {
  * @returns Balances for all wallets
  */
 const useWalletBalances = (): WalletBalanceResult => {
-  const wallets = useWallets();
-
   const nativeCurrency = userAssetsStoreManager().currency;
```

Comment:
Would also move this to a `userAssetsStoreManager.getState().currency` call within the `useMemo`.

This is subscribing to the whole `userAssetsStoreManager` store, not only to `currency`. And allowing `currency` to update here before the wallets summary data when `currency` changes will cause e.g., for a USD → EUR currency change, USD values to briefly render as EUR values given the logic in the `useMemo`.